rllib/models/tf/complex_input_net.py

```diff
@@ -138,16 +166,21 @@ def forward(self, input_dict, state, seq_lens):
                 outs.append(cnn_out)
             elif i in self.one_hot:
                 if "int" in component.dtype.name:
-                    outs.append(
-                        one_hot(component, self.flattened_input_space[i]))
+                    one_hot_in = {
+                        SampleBatch.OBS: one_hot(component,
```

Comment:
we should use SampleBatch instance, instead of a dict here?
same below and for the torch class.

Reply:
+1

Reply:
done